### ISSUE TITLE
DINTO DDI Issues

### ISSUE THREAD

**Issue by c-brenn** [andrewbutterfield/CS4098-2017#1]
There are a few issues with some of the proposed iteration 2 features that we can hopefully iron out.

### DDI Timing Information

The features in the [Timing Analysis Section] assume that DINTO contains information about the timing of DDIs. Unfortunately, there is no such information in the current version(s) of DINTO.

Providing suggestions to users about altering pathway timings, without any idea about DDI time frames, would be irresponsible. Without accurate timing information, the only reasonable suggestion the system can make is that the pathway is unsafe and should be investigated by medical professionals.

### DDI Classification

There is an even more fundamental problem with DINTO. While it is possible to identify a DDI between two drugs, there is no way of knowing whether the interaction would be beneficial to the patient, or if it would be harmful.

There are classes for `potentially harmful` and `potentially beneficial`, but there are no members of these classes. Furthermore, there is no reasonable approach to populating these classes without the assistance of the creators of DINTO. We reached out to the DINTO maintainers several weeks ago about populating these classes, but they have not responded.

Again, it would be irresponsible to assume that any DDI is beneficial. As such - without these updates to DINTO - the system must assume that all DDIs are harmful and report them as such to the users.

## Discussion

These issues affect some of the [timing analysis features][timing analysis section]. Without updates to DINTO, these features in their current form would be unsafe to implement.

I'm opening this issue so there is a central place for discussing these problems - so we can get your input and input from other groups as we work to resolve them.

[timing analysis section]: https://github.com/andrewbutterfield/CS4098-2017/blob/84a0cbfb7931071f5ef8c2ed516f3289e1f6df83/Product-Backlog.md#timing-analysis

**Comment by andrewbutterfield** [comment]
I agree that DINTO hasn't been as useful as I had hoped - but we shall still continue with this approach. It's not unsafe because in order to deploy this in the wild, we'd have to get it certified, and convince real medics to use it - this project is very much at the feasibility study/pre-competitive stage.

What will need to be done is to figure out a way to fake DDI timing information. This also raises the interesting notion that drugs an have self timing constraints. How do we specify in a PML pathway that a drug should be administered at most four times a day with at least four hours between doses? Note that this specification is a bit non-deterministic (e.g. "at most").

**Comment by c-brenn** [comment]
## Fake DDI Timing Information

Adding fake information would certainly help to circumvent some of DINTO issues. This approach requires some group discussion though.

For testing purposes, we probably want the information to be persistent. That is, if you ask for a DDI you always get the same timing information. Inconsistent timing information would make automated testing and acceptance testing more complicated.

This raises a few questions. Should we be modifying DINTO and storing this information there? Should the fake timing data be consistent across groups for your acceptance testing?

## DDI Classification

Similarly, our approach to DDI classification will determine the complexity of some of the release 2 features. As mentioned in my original post, DDIs are not classified as `beneficial` or `harmful` in DINTO.

Are we to assume that all DDIs are harmful, or are groups expected to create fake data regarding DDI classification? 

Both `Identify DDI Closest Approach` and `Identify DDI Furthest Separation` require that we know whether a DDI is harmful or beneficial.

## Documentation

I would suggest that the features in the product backlog be updated to reflect the need to add fake information. Similarly, if we can make assumptions about a DDI classifications, this too should be reflected in the backlog.


> This also raises the interesting notion that drugs an have self timing
  constraints

Again, if this behaviour is expected in the system, it should be noted in the product backlog.


These requirements add significant complexity to the proposed feature set for release 2. My concern is that groups will underestimate the work involved in these features, especially groups that have not had to investigate DINTO very much.

**Comment by mcnestrb** [comment]
On the website for The Obo Foundry (an online group of developers who try to standardise ontologies), they describe DDIs as 

> a significant risk group for adverse effects associated with pharmaceutical treatment

[http://www.obofoundry.org/ontology/dinto.html](http://www.obofoundry.org/ontology/dinto.html)

This seems to suggest that DINTO never identifies beneficial DDIs so I feel that the task `Identify DDI Furthest Separation` can not be completed as that information is not provided by DINTO (unless I am misunderstanding something in which case this is open for discussion)

If this is the case then all DDIs are assumed to be harmful so if we have fake time inserted into DINTO, then `Identify DDI Closest Approach` might be possible

**Comment by andrewbutterfield** [comment]
There are positive interactions discussed in the paper/thesis - but the main concern of clinicians in looking at DDIs is to avoid the adverse ones - hence that emphasis

**Comment by andrewbutterfield** [comment]
A real discussion we need to have is what, if anything, to do about DINTO?

**Comment by shawa** [comment]
Just to sum up and build on discussion in today's lecture, I did a good bit of digging on DINTO last night to try and get to the bottom of this.
Some key points I came across are:

* DINTO is the result of an experiment to apply a number of AI techniques like Natural Language Processing to various corpora of Biomedical Science Literarure to _Infer_ Drug-drug interactions
* DINTO Versions 1 and 1.2 don't differ very much, except that dinto 1.2 contains additonal data from the OEA, the _Ontology of Adverse Events_


[Early slides on DINTO](https://www.slideshare.net/Grupo_LaBDA/dinto-an-ontology-for-drugdrug-interactions) mention a 'Final DINTO' that has all these fields populated.

Here's also a [newer set of slides](https://conferences.ncl.ac.uk/media/sites/conferencewebsites/ukon2016/7-mhz.pdf) (2016)
that mentions under _Future Work_:

* To maintain and update the current version. [This would suggest the project has no current maintainer]
* To include more information from other databases: physicochemical properties,
drug-protein binding affinity, drug therapeutic index.
* To combine inferences with machine learning techniques to **identify clinically relevant DDIs.**

The second point and third points would suggest that as far as the useful information we're looking for goes, DINTO isn't quite there yet.

I have sent an email to María, who came up with DINTO for her PhD, with some of the questions we've come across and will update here if I hear anything back.
